Zend/zend_API.c

```diff
@@ -2318,6 +2323,9 @@ ZEND_API void zend_collect_module_handlers(void) /* {{{ */
 	module_request_shutdown_handlers[shutdown_count] = NULL;
 	module_post_deactivate_handlers = module_request_shutdown_handlers + shutdown_count + 1;
 	module_post_deactivate_handlers[post_deactivate_count] = NULL;
+	/* Cannot reuse module_request_startup_handlers because it is freed in zend_destroy_modules, which happens before zend_unload_modules. */
```

Comment:
It is possible to move freeing module_request_startup_handlers to zend_unload_modules, but that changes public API.
And we want to avoid that in stable branches. The reason for this different workaround for lower branches was to avoid changing the public API in the first place.